packages/flutter/lib/src/widgets/form.dart

```diff
@@ -607,19 +607,19 @@ class FormFieldState<T> extends State<FormField<T>> with RestorationMixin {
 
   @override
   Widget build(BuildContext context) {
+    final FormState? formState = Form.maybeOf(context);
     if (widget.enabled) {
-      switch (widget.autovalidateMode) {
-        case AutovalidateMode.always:
+      final AutovalidateMode? formMode = formState?.widget.autovalidateMode;
+      final AutovalidateMode fieldMode = widget.autovalidateMode;
+      if(formMode == AutovalidateMode.always || fieldMode == AutovalidateMode.always) {
```

Comment:
What if fieldMode is `disabled`? Should that take precedence over formMode?  I think the answer is yes, but the existing form code and docs are pretty unclear about this.

Maybe you could create a getter that gives the effective autovalidateMode considering any parent Form as well?

Reply:
Currently both the `Form` and the `FormField` fallback to `AutovaldiateMode.disabled` as a default. I think that we can take precedence from children over parents, **if** the `autovalidateMode` was explicitly chosen.

However, he current implementation works differently. If a `FormField` has the mode `onUserInteraction` or `disabled`, and the parent `Form` has `always`, the `Form` will "always" run the validation on its registered `FormField`s, even if they have this different mode.

So currently the parent's mode takes precedence over the children's mode. Not sure if we want to change that, even if the docs don't mentioned this specifically.

Reply:
I think we should try to limit the scope of this PR, so let's keep it as close as possible to the original behavior for now (as your PR is) unless there's something clearly wrong.

Though, can you take a look at fixing the test failures? I'm wondering if this change changed any other behavior.

Reply:
I agree, let's keep the current behavior for this.

It was a bit confusing, since I cannot run all tests locally on an M2 chip (https://github.com/flutter/flutter/issues/111739), but after searching the raw logs of the CI tests on Github, it turns out something in the TimePicker is causing a `setState` or `markNeedsBuild` to be called during the build phase. I'll dive into it.

Reply:
I figured out the issue:

`TimePicker` uses a `Form` with 2 `TextFormFields` (hour and minute), which each have their `validator` functions that call its `setState` and sets the `Form`s autovalidateMode to `always`. Because the `FormField` now runs its validators if the parent's mode is not `disabled, it executes the validators, triggering the `TimePicker`'s `setState` during the build phase of the `FormField`.

I assume that would have already been an issue if the `FormField` has `always` or `onUserInteraction`, with a validator that triggers a `setState` 🤔 I'll test this as well, but should a `FormField.validator` function call `setState` in general?

I'll continue on this later in the upcoming days.